pkg/graveler/ref/repository_iterator.go

```diff
@@ -4,99 +4,87 @@ import (
 	"context"
 	"errors"
 
-	"github.com/treeverse/lakefs/pkg/db"
 	"github.com/treeverse/lakefs/pkg/graveler"
+	"github.com/treeverse/lakefs/pkg/kv"
 )
 
-type RepositoryIterator struct {
-	db        db.Database
-	ctx       context.Context
-	value     *graveler.RepositoryRecord
-	buf       []*graveler.RepositoryRecord
-	offset    string
-	fetchSize int
-	err       error
-	state     iteratorState
+type RepositoryIterator interface {
+	Next() bool
+	SeekGE(id graveler.RepositoryID)
+	Value() *graveler.RepositoryRecord
+	Err() error
+	Close()
 }
 
-func NewRepositoryIterator(ctx context.Context, db db.Database, fetchSize int) *RepositoryIterator {
-	return &RepositoryIterator{
-		db:        db,
-		ctx:       ctx,
-		fetchSize: fetchSize,
-		buf:       make([]*graveler.RepositoryRecord, 0, fetchSize),
-	}
+type KVRepositoryIterator struct {
+	ctx   context.Context
+	it    kv.MessageIterator
+	err   error
+	value *graveler.RepositoryRecord
```

Comment:
I chose to use repository data instead of repository record (on the commit), then on the kv iterator you don't need to convert, on the DB iterator you do need, but once we delete the DB the repository record will not be needed anymore. WDYT?

Reply:
me and @niro discussed and decided to keep it with the record for now until we delete the DB and then maybe we'll change it

Reply:
:+1: